src/Microsoft.AspNetCore.OData/Routing/ODataActionSelector.cs

```diff
@@ -95,10 +95,19 @@ public ActionDescriptor SelectBestCandidate(RouteContext context, IReadOnlyList<
         {
             RouteData routeData = context.RouteData;
             ODataPath odataPath = context.HttpContext.ODataFeature().Path;
+
             if (odataPath != null && routeData.Values.ContainsKey(ODataRouteConstants.Action))
             {
+                var odataRoute = routeData.Routers.OfType<ODataRoute>().FirstOrDefault();
```

Comment:
>.FirstOrDefault() [](start = 71, length = 17)

Is it safe to assume that the route data we care about will be the first one? @saxu@microsoft.com; aren't there cases where the user might register multiple odata routes?